app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,18 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(_('Invalid provided user type of %s') % options[:user].class.to_s)
+           end
+
+    unless user.mail_enabled?
+      log.warn(N_("Must specify a user with email enabled"))
+      return
+    end
```

Comment:
I'd rather log this as `debug` because there can be many users in the database possibly all disabled and we run these summaries every day/hour and this can possibly flood logs. Also include user id in the message (not username that's security leak).

Reply:
done